nixos/modules/services/home-automation/matter-server.nix

```
@@ -0,0 +1,123 @@
+{ lib
+, pkgs
+, config
+, ...
+}:
+
+with lib;
+
+let
+  cfg = config.services.matter-server;
+in
+
+{
+  meta.maintainers = with lib.maintainers; [ hexa ];
+
+  options.services.matter-server = with types; {
+    enable = mkEnableOption (lib.mdDoc "Matter-server");
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.python-matter-server;
+      defaultText = literalExpression "pkgs.python-matter-server";
+      description = "Package containing the matter-server executable.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 5580;
+      defaultText = "5580";
+      description = "Port to expose the matter-server service on.";
+    };
+
+    vendorId = mkOption {
+      type = types.ints.u16;
+      default = 4939; # Home Assistant
+      defaultText = "4939 (Home Assistant)";
+      description = "CSA Vendor ID to advertise to devices on the network.";
+    };
+
+    fabricId = mkOption {
+      type = types.int;
+      default = 1;
+      defaultText = "4939 (Home Assistant)";
+      description = "CSA Vendor ID to advertise to devices on the network.";
+    };
+
+    storagePath = mkOption {
+      type = types.nonEmptyStr;
+      default = "matter-server";
+      description = mdDoc "Directory name under `/var/lib/` to store persistent data. Specifying `\"matter-server\"` indicates the directory `/var/lib/matter-server`";
+    };
+
+    logLevel = mkOption {
+      type = types.enum [ "critical" "error" "warning" "info" "debug" ];
+      default = "info";
+      description = "Verbosity of logs from the matter-server";
+    };
+
+    extraArgs = mkOption {
+      type = listOf str;
+      default = [];
+      description = "Extra arguments to pass to the matter-server executable.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.avahi.enable = true;
+
+    systemd.services.matter-server = {
+      after = [
+        "network-online.target"
+        "avahi-daemon.target"
+      ];
+      wantedBy = [
+        "multi-user.target"
+        "home-assistant.target"
```

```suggestion
        "home-assistant.target"
```

Preferably via `wants` and `before`, because if `home-assistant.target` isn't reached, you'll still want `mattter-server.service` to boot up.

---

Done.